### DISCUSSION TITLE
Two copies of the same version of the package

### DISCUSSION THREAD

**Discussion by xiaoxiaojx** [pnpm/pnpm#4051] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/23253540/143979554-083009de-e027-461a-aa85-1c982491e043.png)
I'm more curious about what would cause this，Because it now breaks the singleton pattern, yes the same version of the package's code is run twice。

Thank you for the answer ～

**Comment by zkochan** [comment, marked as answer]
Add webpack as a dependency to your project and this will be solved. The two folders might still remain but the first one will not be symlinked from anywhere.

**Reply by xiaoxiaojx** [reply]
Thanks for the answer, I carefully read the documentation, it should be peer caused by the problem
https://www.pnpm.cn/how-peers-are-resolved

I solved it temporarily with the following solution 😄  

![6003f0787376c585b058dbcf0a9268a64a509efd](https://user-images.githubusercontent.com/23253540/144044010-e504ad2f-5313-407f-af08-79693d5d5e8e.png)

**Reply by meteorlxy** [reply]
In this workaround, we may also need to remove the `pkg.peerDependenciesMeta` if present.

**Comment by Helium-Z** [comment]
Hi @zkochan, I met the same problem. 

And I am sure there is no peerDependency in my package (@arco-cli/service). I compared the difference between the two copies of my package and here is the comparison result. Is there any way to solve this problem? I have tried `pnpm.override` but it doesn't seem to work.

Thanks for your help anyway.

![image](https://user-images.githubusercontent.com/29861850/221150665-96d76a10-f421-44ac-8328-eecfe96e48e5.png)
